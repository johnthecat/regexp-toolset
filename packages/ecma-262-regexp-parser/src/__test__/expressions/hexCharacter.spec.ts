import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Octal char', () => {
  it('should correctly unicode value', () => {
    expect(parseRegexp(/\x61/)).toMatchSnapshot();
  });

  it('should fallback to char sequence, if octal value is out of range', () => {
    // Largest hex value - ff₁₆ (255₁₀) or U+00ff (equals to \u00ff regexp)
    expect(parseRegexp(/\xff/)).toMatchSnapshot('largest value');
    expect(parseRegexp(/\xfg/)).toMatchSnapshot('out of range');
  });

  it('should fallback to char sequence, if there is not enough chars', () => {
    expect(parseRegexp(/\xf\a/)).toMatchSnapshot('with char at the end');
    expect(parseRegexp(/\xf/)).toMatchSnapshot('with regexp end at the end');
  });
});
