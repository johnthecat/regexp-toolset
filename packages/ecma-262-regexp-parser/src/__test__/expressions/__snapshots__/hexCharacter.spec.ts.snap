// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Octal char > should correctly unicode value 1`] = `
{
  "body": {
    "charCode": 97,
    "end": 4,
    "kind": "Char",
    "start": 1,
    "type": "Hex",
    "value": "a",
  },
  "end": 5,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if octal value is out of range > largest value 1`] = `
{
  "body": {
    "charCode": 255,
    "end": 4,
    "kind": "Char",
    "start": 1,
    "type": "Hex",
    "value": "Ã¿",
  },
  "end": 5,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if octal value is out of range > out of range 1`] = `
{
  "body": {
    "end": 4,
    "expressions": [
      {
        "charCode": 120,
        "end": 2,
        "kind": "Char",
        "start": 1,
        "type": "Escaped",
        "value": "x",
      },
      {
        "charCode": 102,
        "end": 3,
        "kind": "Char",
        "start": 3,
        "type": "Simple",
        "value": "f",
      },
      {
        "charCode": 103,
        "end": 4,
        "kind": "Char",
        "start": 4,
        "type": "Simple",
        "value": "g",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 5,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough chars > with char at the end 1`] = `
{
  "body": {
    "end": 5,
    "expressions": [
      {
        "charCode": 120,
        "end": 2,
        "kind": "Char",
        "start": 1,
        "type": "Escaped",
        "value": "x",
      },
      {
        "charCode": 102,
        "end": 3,
        "kind": "Char",
        "start": 3,
        "type": "Simple",
        "value": "f",
      },
      {
        "charCode": 97,
        "end": 5,
        "kind": "Char",
        "start": 4,
        "type": "Escaped",
        "value": "a",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough chars > with regexp end at the end 1`] = `
{
  "body": {
    "end": 3,
    "expressions": [
      {
        "charCode": 120,
        "end": 2,
        "kind": "Char",
        "start": 1,
        "type": "Escaped",
        "value": "x",
      },
      {
        "charCode": 102,
        "end": 3,
        "kind": "Char",
        "start": 3,
        "type": "Simple",
        "value": "f",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;
